src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -30,20 +32,29 @@ def run_escalating_forecast() -> None:
     Run the escalating forecast algorithm on archived until escalating issues.
     """
     logger.info("Starting task for sentry.tasks.weekly_escalating_forecast.run_escalating_forecast")
-    # TODO: Do not limit to project id = 1 and limit 10 once these topics are clarified
-    # TODO: If possible, fetch group_id instead of the entire group model
-    until_escalating_groups = list(
-        Group.objects.filter(
-            status=GroupStatus.IGNORED,
-            substatus=GroupSubStatus.UNTIL_ESCALATING,
-            project__id=1,
-        )[:10]
-    )
-    logger.info(
-        "Checking for archived until escalating groups",
-        extra={"num_groups": len(until_escalating_groups)},
-    )
-    if not until_escalating_groups:
-        return
-
-    generate_and_save_forecasts(until_escalating_groups)
+
+    for project_ids in RangeQuerySetWrapper(
+        Project.objects.filter(status=ObjectStatus.VISIBLE).values_list("id", flat=True),
+        result_value_getter=lambda item: item,
+        step=10000,
+    ):
+        generate_forecasts_for_projects.delay(project_ids=[project_ids])
+
+
+@instrumented_task(
+    name="sentry.tasks.weekly_escalating_forecast.generate_forecasts_for_projects",
+    queue="weekly_escalating_forecast",
+    max_retries=3,
+    default_retry_delay=60,
+)  # type: ignore
+def generate_forecasts_for_projects(project_ids: List[int]) -> None:
+    for project_id in project_ids:
+        for until_escalating_groups in RangeQuerySetWrapper(
+            Group.objects.filter(
+                status=GroupStatus.IGNORED,
+                substatus=GroupSubStatus.UNTIL_ESCALATING,
+                project__id=project_id,
+                last_seen__gte=datetime.now() - timedelta(days=7),
+            )
+        ):
+            generate_and_save_forecasts(groups=[until_escalating_groups])
```

Comment:
Since we're only passing a group at a time here, we'll now make a query to snuba per group rather than batching a few together. Should we read a batch of groups and pass them through together?

Reply:
Wrapping it in a list will fetch the entire query rather than just the batch. I think you want https://docs.python.org/3/library/itertools.html#itertools.islice

Reply:
Tbh what you actually want is https://more-itertools.readthedocs.io/en/stable/_modules/more_itertools/more.html#chunked